src/test/javascript/spec/component/participation-submission/participation-submission.component.spec.ts

```diff
@@ -114,4 +142,76 @@ describe('ParticipationSubmissionComponent', () => {
         fixture.destroy();
         flush();
     }));
+
+    it('Template Submission is correctly loaded', fakeAsync(() => {
+        TestBed.get(ActivatedRoute).params = of({ participationId: 2, exerciseId: 42 });
+        // set all attributes for comp
+        const templateParticipation = new TemplateProgrammingExerciseParticipation();
+        templateParticipation.id = 2;
+        templateParticipation.repositoryUrl = 'https://bitbucket.ase.in.tum.de/scm/ITCPLEASE1/itcplease1-exercise.git';
+        templateParticipation.submissions = [
+            {
+                submissionExerciseType: SubmissionExerciseType.PROGRAMMING,
+                id: 3,
+                submitted: true,
+                type: SubmissionType.MANUAL,
+                submissionDate: moment('2019-07-09T10:47:33.244Z'),
+                commitHash: '123456789',
+                participation: templateParticipation,
+            },
+        ] as ProgrammingSubmission[];
+        const exercise = { type: ExerciseType.PROGRAMMING } as Exercise;
+        stub(exerciseService, 'find').returns(of(new HttpResponse({ body: exercise })));
+        const programmingExercise = { type: ExerciseType.PROGRAMMING, templateParticipation } as ProgrammingExercise;
+        const findWithTemplateAndSolutionParticipationStub = stub(programmingExerciseService, 'findWithTemplateAndSolutionParticipation');
+        findWithTemplateAndSolutionParticipationStub.returns(of(new HttpResponse({ body: programmingExercise })));
+
+        fixture.detectChanges();
+        tick();
+
+        expect(comp.isLoading).to.be.false;
+        expect(findWithTemplateAndSolutionParticipationStub).to.have.been.called;
+        expect(comp.participation).to.be.deep.equal(templateParticipation);
+        expect(comp.submissions).to.be.deep.equal(templateParticipation.submissions);
+
+        // Create correct url for commit hash
+        const commitHashUrl = comp.getCommitUrl(templateParticipation.submissions[0]);
+        expect(commitHashUrl).to.equal('https://bitbucket.ase.in.tum.de/projects/ITCPLEASE1/repos/itcplease1-exercise/commits/123456789');
```

Comment:
Writing this by hand here is not very nice, it should not be something that needs to be changed in two places. Use variables to define why this url must be the result, what it consists of. What is the relation with repositoryUrl?

Reply:
fixed now